[JS] Update migration docs to use path (#7143)
aws-amplify/docs: src/pages/[platform]/build-a-backend/storage/storage-v5-to-v6-migration-guide/index.mdx

This guide will help you migrate Amplify JavaScript v5's Storage APIs to the new v6's Storage APIs.

> ## Removed Options

Reviewer: Isn't deprecated the correct verbage? We haven't removed anything yet wrt to `key` usage

Author: these are for the V5 ones which was removed in v6 and i added another section for deprecated for the key usage ones.

It was suggestion i got from Harshita to make things clear.

Reviewer: Ahh got it, apologies missed that

>Note: We have remapped all capitalized input and output keys to "camelCase" in v6.

## Deprecated Options

> Starting version 6.x.x, the following parameters in input and output options have been marked deprecated and will be removed in a future version.

Reviewer: We'll need to remember to update this version somehow 😅 

```suggestion
Starting with version 6.x.x, the following API parameters and output options have been marked as deprecated and may be removed in a future version.
```

> - `key`: _A new `path` parameter is introduced and will be used instead of `key` parameter in input and output options. The `path` parameter allows for flexible input string or path construction based on user's identity id._

Reviewer: ```suggestion
- `key`: _A new `path` parameter has been introduced and will be used instead of the `key` parameter in input and output options. The `path` parameter allows for flexible path construction based on user's identity id._
```

> - `accessLevel`: _File accessLevel with scoped `public`, `protected` and `private` permission is now deprecated. Instead, [Amplify (Gen2)](https://docs.amplify.aws/gen2/build-a-backend/storage/) introduces a more refined access control mechanism. To offer similar flexibility to clients, we will utilize the `path` parameter for input options._

Reviewer: ```suggestion
- `accessLevel`: _File `accessLevel` with scoped `public`, `protected`, and `private` permissions is now deprecated. Instead, [Amplify (Gen2)](https://docs.amplify.aws/gen2/build-a-backend/storage/) introduces a more refined access control mechanism. To offer similar flexibility to clients, we will utilize the `path` parameter for input options._
```

      ```
      // getUrl
      {
        path: string | ({identityId}: {identityId: string}) => string;
        options?: {
          useAccelerateEndpoint?: boolean;
        };
      }
      ```
      <br/>

>       ```

Reviewer: What's this little guy doing? Ditto below
